Drv/Ip/SocketComponentHelper.cpp

```diff
@@ -0,0 +1,205 @@
+    m_reconnect = reconnect;
+    // Note: the first step is for the IP socket to open the port
+    Os::Task::Arguments arguments(name, SocketComponentHelper::readTask, this, priority, stack, cpuAffinity);
+    Os::Task::Status stat = m_task.start(arguments);
+    FW_ASSERT(Os::Task::OP_OK == stat, static_cast<NATIVE_INT_TYPE>(stat));
+}
+
+SocketIpStatus SocketComponentHelper::startup() {
+    this->m_started = true;
+    return this->getSocketHandler().startup();
+}
+
+bool SocketComponentHelper::isStarted() {
+    bool is_started = false;
+    this->m_lock.lock();
+    is_started = this->m_started;
+    this->m_lock.unlock();
+    return is_started;
+}
+
+SocketIpStatus SocketComponentHelper::open() {
+    SocketIpStatus status = SOCK_FAILED_TO_GET_SOCKET;
+    NATIVE_INT_TYPE fd = -1;
+    this->m_lock.lock();
+    if (not this->m_open) {
+        FW_ASSERT(this->m_fd == -1 and not this->m_open); // Ensure we are not opening an opened socket
+        status = this->getSocketHandler().open(fd);
+        this->m_fd = fd;
+        // Call connected any time the open is successful
+        if (Drv::SOCK_SUCCESS == status) {
+            this->m_open = true;
+            this->m_lock.unlock();
+            this->connected();
+            return status;
+        }
+    }
+    this->m_lock.unlock();
+    return status;
+}
+
+bool SocketComponentHelper::isOpened() {
+    bool is_open = false;
+    this->m_lock.lock();
+    is_open = this->m_open;
+    this->m_lock.unlock();
+    return is_open;
+}
+
+SocketIpStatus SocketComponentHelper::reconnect() {
+    SocketIpStatus status = SOCK_SUCCESS;
+
+    // Handle opening
+
+    // Open a network connection if it has not already been open
+    // TODO: improve this message
+    if (this->isStarted() and (not this->isOpened())) {
+        status = this->open();
+    }
+    return status;
+}
+
+SocketIpStatus SocketComponentHelper::send(const U8* const data, const U32 size) {
+    SocketIpStatus status = SOCK_SUCCESS;
+    this->m_lock.lock();
+    NATIVE_INT_TYPE fd = this->m_fd;
+    this->m_lock.unlock();
+    // Prevent transmission before connection, or after a disconnect
+    if (fd == -1) {
+        status = this->reconnect();
+        // if reconnect wasn't successful, pass the that up to the caller
+        if(status != SOCK_SUCCESS) {
+            return status;
+        }
+        this->m_lock.lock();
+        fd = this->m_fd;
+        this->m_lock.unlock();
+    }
+    status = this->getSocketHandler().send(fd, data, size);
+    if (status == SOCK_DISCONNECTED) {
+        this->close();
+    }
+    return status;
+}
+
+void SocketComponentHelper::shutdown() {
+    this->m_lock.lock();
+    this->getSocketHandler().shutdown(this->m_fd);
+    this->m_started = false;
+    this->m_fd = -1;
+    this->m_lock.unLock();
+}
+
+void SocketComponentHelper::close() {
+    this->m_lock.lock();
+    if (this->m_fd != -1) {
```

Comment:
Probably safe to leave this check out.....as it is allowed to close a `-1` socket.

Reply:
When this close call was in IpSocket it had this protection so I brought it up to SocketComponentHelper. Still think we should remove?

Reply:
It simply does not matter.  We can leave as-is it to cut down on churn.